File: challenges/week2.group-anagrams/ikasty.py
# Runtime: 88 ms, faster than 95.79% of Python3 online submissions
# Memory Usage: 17.9 MB, less than 50.18% of Python3 online submissions
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = defaultdict(list)
        for item in strs:
            result[''.join(sorted(item))].append(item)

        return result.values()

File: challenges/week3.remove-covered-intervals/ikasty.scala
// Runtime: 552 ms, faster than 100.00% of Scala online submissions for Remove Covered Intervals.
// Memory Usage: 53.2 MB, less than 100.00% of Scala online submissions for Remove Covered Intervals.
object Solution {
    def removeCoveredIntervals(intervals: Array[Array[Int]]): Int = intervals
        .sortBy(item => (item(1), -item(0)))
        .foldRight(List[Array[Int]]()) {
            case (item, Nil)                // init condition
                => List(item)
            case (item, list@(head::_))     // add item to front
            if head(0) > item(0) && head(1) > item(1)
                => item :: list
            case (_, list)                  // remove item
                => list
        }.size
}